src/Bicep.Core/Semantics/SemanticModel.cs

```diff
@@ -108,6 +111,15 @@ public SemanticModel(Compilation compilation, BicepSourceFile sourceFile, IFileR
                 return parameters.ToImmutable();
             });
 
+            this.exportedTypesLazy = new(() => ImmutableDictionary.CreateRange(Root.TypeDeclarations.DistinctBy(t => t.Name)
+                // skip over any type without an `@export()` decorator
+                .Where(t => SemanticModelHelper.TryGetDecoratorInNamespace(this,
+                    t.DeclaringType,
+                    SystemNamespaceType.BuiltInName,
+                    LanguageConstants.ExportPropertyName) is not null)
+                .Select(t => new KeyValuePair<string, ExportedTypeMetadata>(t.Name,
+                    new(t.Name, t.Type, DescriptionHelper.TryGetFromDecorator(this, t.DeclaringType))))));
```

Comment:
It seems the `ImmutableDictionary.CreateRange` is not needed if you change `Select` to `ToImmutableDictionary`.